pywt/_extensions/wavelets_list.pxi

```diff
@@ -5,128 +5,156 @@
 __wname_to_code = {
-    "haar": (c"h", 0),
+    "haar": (0, 0),
     
-    "db1": (c"d", 1),
-    "db2": (c"d", 2),
-    "db3": (c"d", 3),
-    "db4": (c"d", 4),
-    "db5": (c"d", 5),
-    "db6": (c"d", 6),
-    "db7": (c"d", 7),
-    "db8": (c"d", 8),
-    "db9": (c"d", 9),
+    "db1": (2, 1),
+    "db2": (2, 2),
+    "db3": (2, 3),
+    "db4": (2, 4),
+    "db5": (2, 5),
+    "db6": (2, 6),
+    "db7": (2, 7),
+    "db8": (2, 8),
+    "db9": (2, 9),
     
-    "db10": (c"d", 10),
-    "db11": (c"d", 11),
-    "db12": (c"d", 12),
-    "db13": (c"d", 13),
-    "db14": (c"d", 14),
-    "db15": (c"d", 15),
-    "db16": (c"d", 16),
-    "db17": (c"d", 17),
-    "db18": (c"d", 18),
-    "db19": (c"d", 19),
+    "db10": (2, 10),
+    "db11": (2, 11),
+    "db12": (2, 12),
+    "db13": (2, 13),
+    "db14": (2, 14),
+    "db15": (2, 15),
+    "db16": (2, 16),
+    "db17": (2, 17),
+    "db18": (2, 18),
+    "db19": (2, 19),
     
-    "db20": (c"d", 20),
-    "db21": (c"d", 21),
-    "db22": (c"d", 22),
-    "db23": (c"d", 23),
-    "db24": (c"d", 24),
-    "db25": (c"d", 25),
-    "db26": (c"d", 26),
-    "db27": (c"d", 27),
-    "db28": (c"d", 28),
-    "db29": (c"d", 29),
+    "db20": (2, 20),
+    "db21": (2, 21),
+    "db22": (2, 22),
+    "db23": (2, 23),
+    "db24": (2, 24),
+    "db25": (2, 25),
+    "db26": (2, 26),
+    "db27": (2, 27),
+    "db28": (2, 28),
+    "db29": (2, 29),
     
-    "db30": (c"d", 30),
-    "db31": (c"d", 31),
-    "db32": (c"d", 32),
-    "db33": (c"d", 33),
-    "db34": (c"d", 34),
-    "db35": (c"d", 35),
-    "db36": (c"d", 36),
-    "db37": (c"d", 37),
-    "db38": (c"d", 38),
+    "db30": (2, 30),
+    "db31": (2, 31),
+    "db32": (2, 32),
+    "db33": (2, 33),
+    "db34": (2, 34),
+    "db35": (2, 35),
+    "db36": (2, 36),
+    "db37": (2, 37),
+    "db38": (2, 38),
     
-    "sym2": (c"s", 2),
-    "sym3": (c"s", 3),
-    "sym4": (c"s", 4),
-    "sym5": (c"s", 5),
-    "sym6": (c"s", 6),
-    "sym7": (c"s", 7),
-    "sym8": (c"s", 8),
-    "sym9": (c"s", 9),
+    "sym2": (3, 2),
+    "sym3": (3, 3),
+    "sym4": (3, 4),
+    "sym5": (3, 5),
+    "sym6": (3, 6),
+    "sym7": (3, 7),
+    "sym8": (3, 8),
+    "sym9": (3, 9),
     
-    "sym10": (c"s", 10),
-    "sym11": (c"s", 11),
-    "sym12": (c"s", 12),
-    "sym13": (c"s", 13),
-    "sym14": (c"s", 14),
-    "sym15": (c"s", 15),
-    "sym16": (c"s", 16),
-    "sym17": (c"s", 17),
-    "sym18": (c"s", 18),
-    "sym19": (c"s", 19),
-    "sym20": (c"s", 20),
+    "sym10": (3, 10),
+    "sym11": (3, 11),
+    "sym12": (3, 12),
+    "sym13": (3, 13),
+    "sym14": (3, 14),
+    "sym15": (3, 15),
+    "sym16": (3, 16),
+    "sym17": (3, 17),
+    "sym18": (3, 18),
+    "sym19": (3, 19),
+    "sym20": (3, 20),
 
-    "coif1": (c"c", 1),
-    "coif2": (c"c", 2),
-    "coif3": (c"c", 3),
-    "coif4": (c"c", 4),
-    "coif5": (c"c", 5),
-    "coif6": (c"c", 6),
-    "coif7": (c"c", 7),
-    "coif8": (c"c", 8),
-    "coif9": (c"c", 9),
+    "coif1": (4, 1),
```

Comment:
For clarity, it might be worth also including the WAVELET_NAME ctypedef in this file as was done in wavelet.pxd, so that these would be:
```"coif2": (COIF, 2), ```
etc.
That would also keep this integers from getting out of sync if the enum in c/wavelets.h changes for some reason in the future.